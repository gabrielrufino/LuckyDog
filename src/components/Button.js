import React, {useMemo} from 'react';
import {StyleSheet, Text, TouchableHighlight} from 'react-native';

export default function Button({flat, label, onPress, outlined}) {
  const styles = useMemo(
    () =>
      StyleSheet.create({
        container: {
          alignItems: 'center',
          borderRadius: 5,
          justifyContent: 'center',
          paddingVertical: 14,
          width: '100%',
          backgroundColor: '#C8AB83',
          ...(outlined
            ? {
                backgroundColor: '#FFFFFF',
                borderColor: '#C8AB83',
                borderWidth: 2,
              }
            : {}),
          ...(flat
            ? {
                backgroundColor: '#FFFFFF',
              }
            : {}),
        },
        label: {
          fontWeight: 'bold',
          color: outlined || flat ? '#C8AB83' : '#FFFFFF',
        },
      }),
    [flat, outlined],
  );

  const underlayColor = useMemo(
    () => (outlined || flat ? '#FFFFFF' : '#CACAAA'),
    [flat, outlined],
  );

  return (
    <TouchableHighlight
      activeOpacity={0.6}
      underlayColor={underlayColor}
      style={styles.container}
      onPress={onPress}>
      <Text style={styles.label}>{label}</Text>
    </TouchableHighlight>
  );
}
