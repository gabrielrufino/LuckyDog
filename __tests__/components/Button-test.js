import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../../src/components/Button';

describe('components/Button.js', () => {
  it('Should renders the default button correctly', () => {
    const label = 'Button label';
    const onPress = jest.fn();

    const tree = renderer
      .create(<Button label={label} onPress={onPress} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should renders the flat button correctly', () => {
    const label = 'Button label';
    const onPress = jest.fn();

    const tree = renderer
      .create(<Button label={label} onPres={onPress} flat />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should renders the outlined button correctly', () => {
    const label = 'Button label';
    const onPress = jest.fn();

    const tree = renderer
      .create(<Button label={label} onPress={onPress} outlined />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
