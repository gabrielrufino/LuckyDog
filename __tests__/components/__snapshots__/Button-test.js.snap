// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button.js Should renders the default button correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#C8AB83",
      "borderRadius": 5,
      "justifyContent": "center",
      "paddingVertical": 14,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontWeight": "bold",
      }
    }
  >
    Button label
  </Text>
</View>
`;

exports[`components/Button.js Should renders the flat button correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "justifyContent": "center",
      "paddingVertical": 14,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#C8AB83",
        "fontWeight": "bold",
      }
    }
  >
    Button label
  </Text>
</View>
`;

exports[`components/Button.js Should renders the outlined button correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#C8AB83",
      "borderRadius": 5,
      "borderWidth": 2,
      "justifyContent": "center",
      "paddingVertical": 14,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#C8AB83",
        "fontWeight": "bold",
      }
    }
  >
    Button label
  </Text>
</View>
`;
